pkg/mqtrigger/scalermanager.go

```diff
@@ -0,0 +1,265 @@
+			go func() {
+				mqt := newObj.(*fv1.MessageQueueTrigger)
+				logger.Debug("Update deployment for Scaler Object", zap.Any("mqt", mqt.ObjectMeta), zap.Any("maqt.Spec", mqt.Spec))
+				scaledObject := getScaledObject(mqt)
+				kedaClient, err := getKedaClient(mqt.ObjectMeta.Namespace)
+				if err != nil {
+					logger.Error("Failed to create KEDA client", zap.Error(err))
+				}
+				_, err = kedaClient.Update(scaledObject, metav1.UpdateOptions{})
+				if err != nil {
+					logger.Error("Failed to Update ScaledObject", zap.Error(err))
+				}
+			}()
+		},
+		DeleteFunc: func(obj interface{}) {
+			go func() {
+				mqt := obj.(*fv1.MessageQueueTrigger)
+				logger.Debug("Delete deployment for Scaler Object", zap.Any("mqt", mqt.ObjectMeta), zap.Any("maqt.Spec", mqt.Spec))
+				name := mqt.ObjectMeta.Name
+				kedaClient, err := getKedaClient(mqt.ObjectMeta.Namespace)
+				if err != nil {
+					logger.Error("Failed to create KEDA client", zap.Error(err))
+				}
+				err = kedaClient.Delete(name, &metav1.DeleteOptions{})
+				if err != nil {
+					logger.Error("Failed to Delete ScaledObject", zap.Error(err))
+				}
+				deletePolicy := metav1.DeletePropagationForeground
+				if err := deploymentsClient.Delete(mqt.ObjectMeta.Name, &metav1.DeleteOptions{
+					PropagationPolicy: &deletePolicy,
+				}); err != nil {
+					logger.Error("Failed to Delete Deployment", zap.Error(err))
+				}
+			}()
+		},
+	})
+	controller.Run(context.Background().Done())
+	return nil
+}
+
+func getScaledObject(mqt *fv1.MessageQueueTrigger) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "ScaledObject",
+			"apiVersion": "keda.k8s.io/v1alpha1",
+			"metadata": map[string]interface{}{
+				"name":      mqt.ObjectMeta.Name,
+				"namespace": mqt.ObjectMeta.Namespace,
+			},
+			"spec": map[string]interface{}{
+				"cooldownPeriod":  &mqt.Spec.CooldownPeriod,
+				"maxReplicaCount": &mqt.Spec.MaxReplicaCount,
+				"minReplicaCount": &mqt.Spec.MinReplicaCount,
+				"pollingInterval": &mqt.Spec.PollingInterval,
+				"scaleTargetRef": map[string]interface{}{
+					"deploymentName": mqt.ObjectMeta.Name,
+				},
+				"triggers": []interface{}{
+					map[string]interface{}{
+						"type":     mqt.ObjectMeta.Name,
+						"metadata": mqt.Spec.Metadata,
+						"authdata": mqt.Spec.Authdata,
+					},
+				},
+			},
+		},
+	}
+}
+
+func getDeploymentSpec(mqt *fv1.MessageQueueTrigger, routerUrl string) *appsv1.Deployment {
+	url := routerUrl + "/" + strings.TrimPrefix(utils.UrlForFunction(mqt.Spec.FunctionReference.Name, mqt.ObjectMeta.Namespace), "/")
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: mqt.ObjectMeta.Name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": mqt.ObjectMeta.Name,
+				},
+			},
+			Template: apiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
```

Comment:
It would be good to add ownerReference to metadata, check https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/

Reply:
Done